### PR TITLE
MM-12755 Add selector for isCommentMention

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Adds a new selector to check if a post should have a highlighted bar based on user pref

#### Ticket Link
[MM-12755](https://mattermost.atlassian.net/browse/MM-12755)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: src/selectors/entities/posts.js [hunk 1]
```diff
@@ -483,3 +467,37 @@ export function getPostIdsInChannel(state, channelId) {
 
 export const isPostIdSending = (state, postId) =>
     state.entities.posts.sendingPostIds.some((sendingPostId) => sendingPostId === postId);
+
+export const makeIsPostCommentMention = () => {
+    return createSelector(
+        getAllPosts,
+        getPostsInThread,
+        getCurrentUser,
+        (state, postId) => getPost(state, postId),
```
Comment (enahum) [inline src/selectors/entities/posts.js]: You can have this just as getPost without the arrow function

Comment (sudheerDev) [reply src/selectors/entities/posts.js]: Yup that's true. Changed it to have just the selector.

File: src/utils/post_utils.js [hunk 2]
```diff
@@ -383,3 +383,26 @@ export function combineSystemPosts(postsIds = [], posts = {}, channelId) {
...
+    }
+    if (currentUser.notify_props && currentUser.notify_props.comments) {
+        commentsNotifyLevel = currentUser.notify_props.comments;
+    }
+
+    const notCurrentUser = post.user_id !== currentUser.id || (post.props && post.props.from_webhook);
+    if (notCurrentUser) {
+        if (commentsNotifyLevel === 'any' && (threadCreatedByCurrentUser || threadRepliedToByCurrentUser)) {
+            isCommentMention = true;
+        } else if (commentsNotifyLevel === 'root' && threadCreatedByCurrentUser) {
```
Comment (saturninoabril) [inline src/utils/post_utils.js]: I think we can use the constants at `constants/preferences`:
```
COMMENTS_ANY: 'any',
COMMENTS_ROOT: 'root',
COMMENTS_NEVER: 'never',
```

Comment (sudheerDev) [reply src/utils/post_utils.js]: Definitely. Made the change.

**Pull request merged**